docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,350 @@
+You need to provide an alternative pattern for all of these `yyyy-MM-dd||yyyy-MM-d`.
+More then 2 digits are not allowed anymore. If you want to parse dates like `2010-01-00001` you need to create a pattern that expects a text literal `'000'` before `'d'`
+`yyyy-MM-'000'dd`
+--
+
+
+`e` name of day::
++
+--
+The same meaning in 7.0. However it is less flexible in 7.0. It do not allow parsing short and full text forms.
+
+If you used `EEE YYYY-MM` for a format and expected days in form:
+`Wed 2020-01` or `Wednesday 2020-01`
+then you have to use two combined patterns in 7.0
+`cccc yyyy-MM||ccc yyyy-MM`
+
+E vs c. E means day-of-week vs c is for localized day-of-week. The difference is that E won't behave correctly for full text forms like Wednesday
+--
+
+Text forms EEEE and similar::
++
+--
+Full text forms are depending on locale data provided with JDK and the implementation details of java vs joda. You should test carefully before upgrading these patterns.
+--
+
+`z` time zone text::
++
+--
+In 7.0 it will print 'Z' for Zulu given UTC timezone.
+--
+
+
+===== Test with your data
+All of the examples above should be tested with real data. It is likely that you do not need the flexibility joda was giving.
+If you expect your timezone to only be in one form - for instance `+01:00`  (XXX in java, ZZ in joda) -
+then there is no need to create a new java pattern with so many alternatives.
+
+Consider using this date debugging site for assistance https://esddd.herokuapp.com/
+
+==== Migrating affected mappings
+Once an index is created with a mapping, it cannot change already existing fields. You need to create a new index with updated mapping
+and reindex your data to it.
+You can however update your pipelines or templates. Remember to look for templates you use with tools outside ES.
+If you specified a custom date format there, then you need to update it too.
+
+===== Example migration procedure
+Let's assume that you have an index with a date field and custom format
+////
+[source,console]
+--------------------------------------------------
+PUT my_index_1
+{
+    "mappings" : {
+      "properties" : {
+         "datetime": {
+           "type": "date",
+           "format": "yyyy/MM/dd HH:mm:ss||yyyy/MM/dd||epoch_millis"
+         }
+      }
+    }
+}
+--------------------------------------------------
+////
+
+[source,console]
+--------------------------------------------------
+GET my_index_1/_mapping
+--------------------------------------------------
+// TEST[continued]
+
+[source,console-result]
+--------------------------------------------------
+{
+  "my_index_1" : {
+    "mappings" : {
+      "properties" : {
+         "datetime": {
+           "type": "date",
+           "format": "yyyy/MM/dd HH:mm:ss||yyyy/MM/dd||epoch_millis"
+         }
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+
+* Create my_index_2 with mappings changed with the format set to `8uuuu/MM/dd HH:mm:ss||uuuu/MM/dd||epoch_millis`
```

Comment:
To change the date format for the `datetime` field, create a separate index
containing an updated mapping and date format.

For example, the following `my_index_2` index changes the date format to `8uuuu/MM/dd
HH:mm:ss||uuuu/MM/dd||epoch_millis`. The `8` prefix indicates this data format
use java time.